andela/ah-frontend-poseidon
# #162162962 Implement sign-up form

#### What does this PR do?
- Implement sign-up form with live validations.
- Implement a spinner component.
- Add a notification component  
- Write tests for the signup form.
#### Description of Task to be completed?
- A user should be able to access a well styled up form for signup.
- A user should be able to view error messages in-case invalid data is input.
- Users should be able to Signup and or Register using Sign up forms on the signup route.
#### How should this be manually tested?
After cloning the repo, cd into it and follow the steps 
* In terminal, type `git checkout ft-sign-up-form-162162962` to switch to my branch.
* Then type `yarn add` to install all the project dependencies in package.json **Note** ensure that you have nodejs installed on your computer. 
* Finally, type `yarn start` to run the project server and then open your browser and type `http://localhost:3000/signup` to access the signup page

#### What are the relevant pivotal tracker stories?
[#162162962](https://www.pivotaltracker.com/story/show/162162962)
#### Screenshots
**Empty sign-up form**
<img width="1438" alt="screen shot 2019-01-17 at 10 03 13" src="https://user-images.githubusercontent.com/25249904/51301499-826e3300-1a40-11e9-904d-93c2e204b7fd.png">
**Sign-up form with some data**
<img width="1440" alt="screen shot 2019-01-17 at 10 08 23" src="https://user-images.githubusercontent.com/25249904/51301506-89954100-1a40-11e9-8880-32ba3892ae2b.png">
**Response on successfull sign-up**
<img width="1439" alt="screen shot 2019-01-17 at 10 08 50" src="https://user-images.githubusercontent.com/25249904/51301509-8c903180-1a40-11e9-9071-9a5c8089178d.png">
**Response on unsuccessfull sign-up**
<img width="1437" alt="screen shot 2019-01-17 at 10 21 50" src="https://user-images.githubusercontent.com/25249904/51301905-cb72b700-1a41-11e9-845b-9a113f65c283.png">
